server/src/main/java/org/elasticsearch/common/filesystem/WindowsFileSystemNatives.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.filesystem;
+
+import com.sun.jna.Native;
+import com.sun.jna.WString;
+import com.sun.jna.ptr.IntByReference;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.util.Constants;
+import org.elasticsearch.core.Nullable;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * {@link FileSystemNatives.Provider} implementation for Windows/Kernel32
+ */
+final class WindowsFileSystemNatives implements FileSystemNatives.Provider {
+
+    private static final Logger logger = LogManager.getLogger(WindowsFileSystemNatives.class);
+
+    private static final class Holder {
+        private static final WindowsFileSystemNatives instance = new WindowsFileSystemNatives();
+    }
```

Comment:
I wonder if this lazy holder idiom is really needed here? Or could it just be replaced with
```
  private static final WindowsFileSystemNatives INSTANCE = new WindowsFileSystemNatives();
```

Reply:
I don't think it is really needed so I simplified it.